arroyo/types.py

```diff
@@ -39,13 +46,17 @@ class Message(Generic[TPayload]):
 
     __slots__ = ["value"]
 
-    value: BaseValue[TPayload]
+    value: BaseValue[Union[FilteredPayload, TPayload]]
 
     def __init__(
         self,
-        value: BaseValue[TPayload],
+        value: Union[
+            BaseValue[Union[FilteredPayload, TPayload]],
+            BaseValue[FilteredPayload],
+            BaseValue[TPayload],
```

Comment:
Not sure I get why we need all three cases.

Reply:
I added a code comment